api/controllers/StatsController.js

```diff
@@ -118,18 +180,34 @@ module.exports = {
             // Find season this match took place during
             return dayjs(match.startTime).isBetween(dayjs(season.startTime), dayjs(season.endTime));
           });
-          if (relevantSeason) {
-            const player1 = idToUserMap.get(match.player1);
-            const player2 = idToUserMap.get(match.player2);
-            if (player1 && player2) {
-              // Player 1
-              addMatchToRankings(relevantSeason, match, player1, player2);
-              // Player 2
-              addMatchToRankings(relevantSeason, match, player2, player1);
-            }
+          if (!relevantSeason) {
+            return;
+          }
+          const player1 = idToUserMap.get(match.player1);
+          const player2 = idToUserMap.get(match.player2);
+          if (player1 && player2) {
+            // Player 1
+            addMatchToRankings(relevantSeason, match, player1, player2);
+            // Player 2
+            addMatchToRankings(relevantSeason, match, player2, player1);
           }
         }
       });
+
+      games.forEach((game) => {
+        const relevantSeason = seasons.find((season) => {
+          // Find season this match took place during
+          return dayjs(game.updatedAt).isBetween(dayjs(season.startTime), dayjs(season.endTime));
+        });
```

Comment:
It might make sense to pull the games update and the match updates into their own functions to make the actual controller endpoint a bit more lean, idk. @seriouslysean how clearly does this read to you?